charts/apisix/templates/ingress.yaml

```diff
@@ -34,6 +34,9 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  {{- if .Values.gateway.ingress.className }}
+  ingressClassName: {{ .Values.gateway.ingress.className }}
```

Comment:
This field was introduced since k8s 1.18, but the ingress resource for apisix is also compatible with older ingress version (extensions). So only add this field only if the k8s version is new enough.

We also need to consider removing the support for older ingress versions.

Reply:
Condition modified to the suggestion.